src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/LatencyAndInflightCountLoadBalancingPolicy.java

```diff
@@ -0,0 +1,289 @@
+      if (nbMeasure < thresholdToAccount)
+        return new TimestampedAverage(currentTimestamp, -1L, nbMeasure);
+
+      if (previous == null || previous.average < 0)
+        return new TimestampedAverage(currentTimestamp, newLatencyNanos, nbMeasure);
+
+      // Note: it's possible for the delay to be 0, in which case newLatencyNanos will basically be
+      // discarded. It's fine: nanoTime is precise enough in practice that even if it happens, it
+      // will be very rare, and discarding a latency every once in a while is not the end of the
+      // world.
+      // We do test for negative value, even though in theory that should not happen, because it
+      // seems
+      // that historically there has been bugs here
+      // (https://blogs.oracle.com/dholmes/entry/inside_the_hotspot_vm_clocks)
+      // so while this is almost surely not a problem anymore, there's no reason to break the
+      // computation
+      // if this even happen.
+      long delay = currentTimestamp - previous.timestamp;
+      if (delay <= 0) return null;
+
+      double scaledDelay = ((double) delay) / scale;
+      // Note: We don't use log1p because we it's quite a bit slower and we don't care about the
+      // precision (and since we
+      // refuse ridiculously big scales, scaledDelay can't be so low that scaledDelay+1 == 1.0 (due
+      // to rounding)).
+      double prevWeight = Math.log(scaledDelay + 1) / scaledDelay;
+      long newAverage =
+          (long) ((1.0 - prevWeight) * newLatencyNanos + prevWeight * previous.average);
+
+      return new TimestampedAverage(currentTimestamp, newAverage, nbMeasure);
+    }
+
+    public TimestampedAverage getCurrentAverage() {
+      return current.get();
+    }
+  }
+
+  private void compareAndSwapSlowReplica(Object[] currentNodes, int i, int j) {
+    NodeLatencyTracker tracker1 = latencies.getUnchecked((Node) currentNodes[i]);
+    NodeLatencyTracker tracker2 = latencies.getUnchecked((Node) currentNodes[j]);
+    if (tracker1 != null && tracker2 != null) {
+      TimestampedAverage average1 = tracker1.getCurrentAverage();
+      TimestampedAverage average2 = tracker2.getCurrentAverage();
+      if (average1 != null
+          && average2 != null
+          && average1.average > average2.average
+          && System.nanoTime() - average1.timestamp < RETRY_PERIOD) {
+        ArrayUtils.swap(currentNodes, i, j);
+      }
+    }
+  }
+
+  protected int getInFlight(@NonNull Node node, @NonNull Session session) {
+    // The cast will always succeed because there's no way to replace the internal session impl
+    ChannelPool pool = ((DefaultSession) session).getPools().get(node);
+    // Note: getInFlight() includes orphaned ids, which is what we want as we need to account
+    // for requests that were cancelled or timed out (since the node is likely to still be
+    // processing them).
+    return (pool == null) ? 0 : pool.getInFlight();
+  }
```

Comment:
We don't need this implementation here, do we?  The method is already declared (with protected access) [in the superclass](https://github.com/apache/cassandra-java-driver/blob/4.18.1/core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicy.java#L313-L320) with the same implementation.  Can't we just use that?

Reply:
Oh I think it's my legacy code when I tried to make it compatible with even earlier versions. Deleted.